client/my-sites/checkout/composite-checkout/types/transaction-endpoint.ts

```diff
@@ -130,6 +137,34 @@ export function createTransactionEndpointCartFromLineItems( {
 	};
 }
 
+function createTransactionEndpointCartItemFromLineItem(
+	item: WPCOMCartItem
+): WPCOMTransactionEndpointCartItem {
+	return {
+		product_id: item.wpcom_meta?.product_id,
+		meta: item.wpcom_meta?.meta,
+		currency: item.amount.currency,
+		volume: item.wpcom_meta?.volume ?? 1,
+		extra: item.wpcom_meta?.extra,
+	} as WPCOMTransactionEndpointCartItem;
+}
+
+function addRegistrationDataToGSuiteItem(
+	item: WPCOMCartItem,
+	contactDetails: WPCOMTransactionEndpointDomainDetails
+): WPCOMCartItem {
+	if ( ! item.wpcom_meta?.extra.google_apps_users ) {
+		return item;
+	}
+	return {
+		...item,
+		wpcom_meta: {
+			...item.wpcom_meta,
+			extra: { ...item.wpcom_meta.extra, google_apps_registration_data: contactDetails },
+		},
+	} as WPCOMCartItem;
```

Comment:
We should probably filter out some properties such as `extra` and `fax` that are specific to domains. Here is an example of what the API endpoint currently receives:

```php
'google_apps_registration_data' =>
  array (
    'firstName' => 'Gary',
    'lastName' => 'Cooper',
    'organization' => '',
    'email' => 'email@example.com',
    'alternateEmail' => 'another@domain.com',
    'phone' => '+33.123445677',
    'address1' => 'Somewhere',
    'address2' => '',
    'city' => 'Somewhere',
    'state' => '',
    'postalCode' => '12345',
    'countryCode' => 'FR',
    'fax' => '',
    'extra' =>
    array (
      'ca' =>
      array (
        'ciraAgreementAccepted' => false,
      ),
      'uk' =>
      array (
      ),
      'fr' =>
      array (
        'registrantVatId' => '',
      ),
    ),
  ),
```

Reply:
That's a fair point, but Typescript is complex and getting this right is tricky. Is this a blocker for sending this property?

Reply:
No, it's certainly not 👍.

Reply:
Thanks! I've created [a card](https://github.com/Automattic/wp-calypso/projects/89#card-40935778) to track this so it doesn't get forgotten. I'm sure we can get it done soon.